lib/index.js

```diff
@@ -1723,38 +1749,35 @@ exposeMethod('pluck');
  */
 
 Stream.prototype.pickBy = function (f) {
-
-    return this.consume(function (err, x, push, next) {
+    return this.map(objectOnly(function (x) {
         var out = {};
-        if (err) {
-            push(err);
-            next();
+        var seen = {};  // prevents testing overridden properties multiple times.
```

Comment:
Doesn't `Object.create` fall prey to same problem as `Object.getPrototypeOf` in that IE8 doesn't support it?